src/test/java/com/maxmind/db/ReaderTest.java

```diff
@@ -78,6 +78,317 @@ static class GetRecordTest {
+        new networkTest(
+            "0.0.0.0",
+            0,
+            "ipv4",
+            new String[]{
+                "1.1.1.1/32",
+                "1.1.1.2/31",
+                "1.1.1.4/30",
+                "1.1.1.8/29",
+                "1.1.1.16/28",
+                "1.1.1.32/32",
+            }
+        ),
+        new networkTest(
+            "1.1.1.1",
+            30,
+            "ipv4",
+            new String[]{
+                "1.1.1.1/32",
+                "1.1.1.2/31",
+            }
+        ),
+        new networkTest(
+            "1.1.1.1",
+            32,
+            "ipv4",
+            new String[]{
+                "1.1.1.1/32",
+            }
+        ),
+        new networkTest(
+            "255.255.255.0", 
+            24,
+            "ipv4",
+            new String[]{}
+        ),
+        new networkTest(
+            "1.1.1.1",
+            32,
+            "mixed",
+            new String[]{
+                "1.1.1.1/32",
+            }
+        ),
+        new networkTest(
+            "255.255.255.0",
+            24,
+            "mixed",
+            new String[]{}
+        ),
+        new networkTest(
+            "::1:ffff:ffff",
+            128,
+            "ipv6",
+            new String[]{
+                "0:0:0:0:0:1:ffff:ffff/128",
+            },
+            true
+        ),
+        new networkTest(
+            "::",
+            0,
+            "ipv6",
+            new String[]{
+                "0:0:0:0:0:1:ffff:ffff/128",
+                "0:0:0:0:0:2:0:0/122",
+                "0:0:0:0:0:2:0:40/124",
+                "0:0:0:0:0:2:0:50/125",
+                "0:0:0:0:0:2:0:58/127",
+            }
+        ),
+        new networkTest(
+            "::2:0:40",
+            123,
+            "ipv6",
+            new String[]{
+                "0:0:0:0:0:2:0:40/124",
+                "0:0:0:0:0:2:0:50/125",
+                "0:0:0:0:0:2:0:58/127",
+            }
+        ),
+        new networkTest(
+            "0:0:0:0:0:ffff:ffff:ff00",
+            120,
+            "ipv6",
+            new String[]{}
+        ),
+        new networkTest(
+            "0.0.0.0",
+            0,
+            "mixed",
+            new String[]{
+                "1.1.1.1/32",
+                "1.1.1.2/31",
+                "1.1.1.4/30",
+                "1.1.1.8/29",
+                "1.1.1.16/28",
+                "1.1.1.32/32",  
+            }
+        ),
+        new networkTest(
+            "0.0.0.0",
+            0,
+            "mixed",
+            new String[]{
+                "1.1.1.1/32",
+                "1.1.1.2/31",
+                "1.1.1.4/30",
+                "1.1.1.8/29",
+                "1.1.1.16/28",
+                "1.1.1.32/32",  
+            },
+            true
+        ),
+        new networkTest(
+            "::",
+            0,
+            "mixed",
+            new String[]{
+                // ::/96
+                "1.1.1.1/32",
```

Comment:
This is right based on the database specification, but the ones at `::ffff:0:0/96` should look like `::ffff:1.1.1.1/128` or  `::ffff:101:101/128`. This is what I was referring to when we talked.

However, I just looked at the `InetAddress` docs, and it looks like this has behavior similar to `net.IP` in Go:

> In InetAddress and Inet6Address, it is used for internal representation; it has no functional role. Java will never return an IPv4-mapped address. These classes can take an IPv4-mapped address as input, both in byte array and text representation. However, it will be converted into an IPv4 address.

Given that it isn't possible to return `::ffff:1.1.1.1` or whatever, I suppose it would be best to just embrace it. I think we should switch it back to something more similar to what you had. Sorry about that!